src/backend/database_migrations/versions/20210728_213140_backfill_gisaid_epi_ids.py

```diff
@@ -0,0 +1,1510 @@
+    "USA/CA-CZB-33687/2020": "EPI_ISL_2497810",
+    "USA/CA-CZB-33689/2020": "EPI_ISL_2497744",
+    "USA/CA-CZB-33690/2020": "EPI_ISL_2497712",
+    "USA/CA-CZB-33691/2020": "EPI_ISL_2497736",
+    "USA/CA-CZB-33692/2020": "EPI_ISL_2497804",
+    "USA/CA-CZB-33693/2020": "EPI_ISL_2497819",
+    "USA/CA-CZB-33694/2020": "EPI_ISL_2497802",
+    "USA/CA-CZB-33695/2020": "EPI_ISL_2497829",
+    "USA/CA-CZB-33696/2021": "EPI_ISL_2497574",
+    "USA/CA-CZB-33697/2021": "EPI_ISL_2497573",
+    "USA/CA-CZB-33698/2021": "EPI_ISL_2497619",
+    "USA/CA-CZB-33699/2021": "EPI_ISL_2496838",
+    "USA/CA-CZB-33700/2021": "EPI_ISL_2658684",
+    "USA/CA-CZB-33703/2021": "EPI_ISL_2496810",
+    "USA/CA-CZB-33749/2021": "EPI_ISL_2658735",
+    "USA/CA-CZB-33750/2021": "EPI_ISL_2496941",
+    "USA/CA-CZB-33751/2021": "EPI_ISL_2658598",
+    "USA/CA-CZB-33755/2021": "EPI_ISL_2496845",
+    "USA/CA-CZB-33757/2021": "EPI_ISL_2496895",
+    "USA/CA-CZB-33758/2021": "EPI_ISL_2496788",
+    "USA/CA-CZB-33761/2021": "EPI_ISL_2496787",
+    "USA/CA-CZB-33765/2021": "EPI_ISL_2496777",
+    "USA/CA-CZB-33767/2021": "EPI_ISL_2496776",
+    "USA/CA-CZB-33771/2021": "EPI_ISL_2496976",
+    "USA/CA-CZB-33773/2021": "EPI_ISL_2496931",
+    "USA/CA-CZB-33782/2021": "EPI_ISL_2496923",
+    "USA/CA-CZB-33788/2021": "EPI_ISL_2658592",
+    "USA/CA-CZB-33789/2021": "EPI_ISL_2496892",
+    "USA/CA-CZB-33790/2021": "EPI_ISL_2658589",
+    "USA/CA-CZB-33794/2021": "EPI_ISL_2496822",
+    "USA/CA-CZB-3905/2020": "EPI_ISL_2758546",
+    "USA/CA-CZB-4734/2020": "EPI_ISL_2758547",
+    "USA/CA-CZB-4857/2020": "EPI_ISL_2658572",
+    "USA/CA-CZB-5173/2020": "EPI_ISL_2659537",
+    "USA/CA-CZB-5679/2020": "EPI_ISL_2658660",
+    "USA/CA-CZB-5697/2020": "EPI_ISL_2658661",
+    "USA/CA-CZB-6164/2020": "EPI_ISL_2941520",
+    "USA/CA-CZB-6723/2020": "EPI_ISL_2659553",
+    "USA/CA-CZB-6724/2020": "EPI_ISL_2658691",
+    "USA/CA-CZB-6812/2020": "EPI_ISL_2758548",
+    "USA/CA-CZB-7127/2020": "EPI_ISL_2659547",
+    "USA/CA-CZB-7144/2020": "EPI_ISL_2658743",
+    "USA/CA-CZB-7180/2020": "EPI_ISL_2659605",
+    "USA/CA-CZB-7182/2020": "EPI_ISL_2659587",
+    "USA/CA-CZB-7192/2020": "EPI_ISL_2659250",
+    "USA/CA-CZB-7200/2020": "EPI_ISL_2659523",
+}
+
+
+def upgrade():
+    bind = op.get_bind()
+    session = Session(bind=bind)
+    samples_to_update = (
+        session.query(Sample)
+        .filter(Sample.public_identifier.in_(MISSING_EPI_ISL_MAPPING.keys()))
+        .options(
+            joinedload(Sample.uploaded_pathogen_genome),
+        )
+        .all()
+    )
+    for sample in samples_to_update:
+        epi_isl = MISSING_EPI_ISL_MAPPING[sample.public_identifier]
+        sample.uploaded_pathogen_genome.add_accession(
+            repository_type=PublicRepositoryType.GISAID,
+            public_identifier=epi_isl,
+            workflow_start_datetime=datetime.datetime.now(),
+            workflow_end_datetime=datetime.datetime.now(),
+        )
+    session.commit()
```

Comment:
so i think when writing migrations we should only be using raw sql because otherwise if our model api changes then this migration will fail (or at least that's what tony told me awhile ago)